recipes/libcusolver/meta.yaml

```diff
@@ -0,0 +1,185 @@
+{% set name = "libcusolver" %}
+{% set version = "11.4.2.57" %}
+{% set cuda_version = "12.0" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set target_name = "x86_64-linux" %}  # [linux64]
+{% set target_name = "ppc64le-linux" %}  # [ppc64le]
+{% set target_name = "sbsa-linux" %}  # [aarch64]
+{% set target_name = "x64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: c9460db500d0c7d14debb2c8e0d88fc26a39197363f1b000ccf760866f636f12  # [linux64]
+  sha256: bf3c9a090652d011aafe2190c6f26fe108194cf8512d3c571e406054740168bb  # [ppc64le]
+  sha256: 0ca46b73520657f96f921711c05af7053bd398328d510aecddfff2521c000e91  # [aarch64]
+  sha256: c09836df1f43a3496f2b61ad8827af662ace298508869564a40c30d909ba550f  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+outputs:
+  - name: libcusolver
+    files:
+      - lib/libcusolver*.so.*                            # [linux]
+      - targets/{{ target_name }}/lib/libcusolver*.so.*  # [linux]
+      - Library\bin\cusolver*.dll                        # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - sysroot_{{ target_platform }} 2.17    # [linux]
+      host:
+        - cuda-version {{ cuda_version }}
+        - libcublas
+        - libnvjitlink
```

Comment:
Should these be versioned more explicitly? I get that the `cuda-version` package will constrain them to a specific minor version, but I think we'd be best off explicitly capturing the dependency versions instead of relying purely on the `cuda-version` package.

Reply:
It is @adibbley's intention (and also kinda by design) to avoid explicitly spell out the version, and let conda do the work for us. Otherwise, it's too much maintenance overhead, even with automation.

Reply:
Yeah I would much rather let `cuda-version` handle the versioning here. It is common for a CUDA update release to not bump all component versions. 

So we could have a situation where CUDA 12.0.X and CUDA 12.0.Y ship the same version of `libcusolver`, but a new `libcublas`. Leaving these unversioned gives us the `>12.0,<12.1` for free. This constraint would be error prone with every component following their own versioning.